src/Avalonia.Input/MouseDevice.cs

```diff
@@ -104,6 +104,18 @@ public void ProcessRawEvent(RawInputEventArgs e)
                 ProcessRawEvent(margs);
         }
 
+        public void LayoutUpdated(IInputRoot root)
+        {
+            if (Captured == null)
```

Comment:
It feels like we can be a bit more intelligent here. Currently, if I understand correctly we're recalculating the pointer-over state on every invalidation of the scene: that's going to happen _really_ often. We should be able to tell if the scene invalidation affects the point where the mouse currently is, right?

Reply:
that would be great, but is not an area of avalonia I have much of a feel for yet.

Reply:
we could pass the invalidated rect and compare to the last mouse pos, there is also the issue of the window moving to think of.

Reply:
Yeah that's along the lines of what I was thinking. Isn't the pointer position already in windows coordinates, so moving windows shouldn't have an effect?

Reply:
if the window moves by some means other than mouse interaction , the mouse will effectively move relative to the window without mousemove firing and I'm not sure if the scene is invalidated. Not a common thing :)